### DISCUSSION TITLE
Supported Versions of Pytest & Deprecation

### DISCUSSION THREAD

**Discussion by eleanorjboyd** [pytest-dev/pytest#12253] [Q&A]
What versions does the pytest team support? Similar to https://devguide.python.org/versions/ I am looking to find which version is the lowest compatible version that the pytest team maintains security fixes on. 

I am asking since I manage testing support in the python for vscode extension and want to align our pytest support with what the community still supports. I tried looking on your docs but didn't find anything so if you have a place I should look for updates on this that would be very helpful too as I could check back in the future to update the extensions compatibility. 

This came up for me because I am running into an issue with pytest 6.2.5 still having `pytest.Instance` while it is deprecated in pytest 7.0.0.

Thanks for your help!

**Comment by The-Compiler** [comment]
As far as I know, we only really officially support the latest release, there aren't any LTS branches. Note that pytest itself has not had a (known) security vulnerability so far.

**Comment by eleanorjboyd** [comment]
Thank you for clarification! Any thoughts on which one makes sense from your perspective? Since we only support 3.7+ on the extension (because of the python community) maybe 7.1+? I want to give a little wiggle room as our users have a variety of constraints that make updating hard but I have struggled to support versions below 7 as this does come up a few times with people [like this issue](https://github.com/microsoft/vscode-python/issues/22457)

**Reply by RonnyPfannschmidt** [reply]
i'd say it depends on what you actually need/want to support

numerous pytest plugins that want to support a wide range of pytest versions do have version checks for certain code paths

**Reply by The-Compiler** [reply]
Personally I'd indeed go for 7.0.0 as a lower bound. It had major API changes making cross-compatibility for plugins difficult for example, and e.g. the popular pytest-xdist plugin also requires 7.0.0 as of recently: https://github.com/pytest-dev/pytest-xdist/issues/1057 (and so do [pytest-asyncio](https://github.com/pytest-dev/pytest-asyncio) or [pytest-django](https://github.com/pytest-dev/pytest-django/blob/master/pyproject.toml)).

Time-wise, 7.0.0 was released in February 2022, which should also be enough time for people to adapt. The other somewhat obvious choice would be 8.0.0, but that's much more recent (January 2024), and I'd say has less worrying breaking changes. At least from gut feeling, 7.0.0 was the much bigger break.
